### PR TITLE
COMPAT: numpy compat with NaT != NaT, #12049

### LINKED ISSUE #12049: BLD: numpy master changes breaking
(24 hrs ago) good build: https://travis-ci.org/pydata/pandas/jobs/102356098: 1.11.0.dev0+51d2ecd
(1 hr ago) breaking lots of things: https://travis-ci.org/pydata/pandas/jobs/102596904: 1.11.0.dev0+aa6335c

@shoyer IIRC a couple of your PR's were merged in the last day.

here's an example:

```
======================================================================
FAIL: test_coercion_with_setitem_and_series (pandas.tests.test_indexing.TestSeriesNoneCoercion)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/pydata/pandas/pandas/tests/test_indexing.py", line 5247, in test_coercion_with_setitem_and_series
    expected_series.values, strict_nan=True)
  File "/home/travis/build/pydata/pandas/pandas/util/testing.py", line 866, in assert_numpy_array_equal
    raise_assert_detail(obj, msg, left, right)
  File "/home/travis/build/pydata/pandas/pandas/util/testing.py", line 825, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: numpy array are different
numpy array values are different (33.33333 %)
[left]:  [NaT, 2000-01-02T00:00:00.000000000+0000, 2000-01-03T00:00:00.000000000+0000]
[right]: [NaT, 2000-01-02T00:00:00.000000000+0000, 2000-01-03T00:00:00.000000000+0000]
```

### PR DESCRIPTION
**Pull request by jreback**
closes #12049

### CODE REVIEW

**Comment by jreback** [comment]
cc @shoyer 

so just a minor change actually. So ok from me to leave numpy as is I think. (though you still might want to deprecate / do next release in any event)

File: pandas/core/common.py [hunk 1]
```diff
@@ -386,13 +386,21 @@ def array_equivalent(left, right, strict_nan=False):
...
-    if issubclass(left.dtype.type, (np.floating, np.complexfloating)):
+    if is_float_dtype(left) or is_complex_dtype(left):
         return ((left == right) | (np.isnan(left) & np.isnan(right))).all()
 
     # numpy will will not allow this type of datetimelike vs integer comparison
     elif is_datetimelike_v_numeric(left, right):
         return False
 
+    # M8/m8
+    elif needs_i8_conversion(left) or needs_i8_conversion(right):
```
Comment (shoyer) [inline pandas/core/common.py]: we also need to check dtypes -- both timedelta or both datetime

**Comment by shoyer** [comment]
I think that people will be unhappy if numpy issues a release that breaks the latest release of pandas.... so I think numpy needs to wait either way.

File: pandas/core/common.py (hunk 1 above)
Comment (jreback) [inline pandas/core/common.py]: right ok!

**Comment by jreback** [comment]
ok should be good now.

if its changed to a ``DeprecationWarning`` I *think* we shouldn't show it, but will have to see

File: pandas/tests/test_common.py [hunk 2]
```diff
@@ -322,20 +324,40 @@ def test_array_equivalent():
                             np.array([np.nan, 1, np.nan]))
     assert array_equivalent(np.array([np.nan, None], dtype='object'),
                             np.array([np.nan, None], dtype='object'))
-    assert array_equivalent(np.array([np.nan, 1+1j], dtype='complex'),
```
Comment (shoyer) [inline pandas/tests/test_common.py]: glad to see some pep8 clean up :)

**Comment by shoyer** [comment]
LGTM

**Comment by jreback** [comment]
now if Travis-CI would just work :< notice its been red all day

File: pandas/tests/test_common.py (hunk 2 above)
Comment (jreback) [inline pandas/tests/test_common.py]: I have gotten religion err ``flake8``

**Pull request merged**